src/gc-arena/src/types.rs

```diff
@@ -48,10 +61,41 @@ impl GcFlags {
         self.0.get() & 0x4 != 0x0
     }
 
+    /// This is `true` if we've traced a weak pointer during to this `GcBox`
+    /// during the most recent `Phase::Propagate`. This is reset back to
+    /// `false` during `Phase::Sweep`.
+    #[inline]
+    pub(crate) fn has_weak_ref(&self) -> bool {
+        self.0.get() & 0x8 != 0x0
+    }
+
+    /// Determines whether or not we've dropped the stored `dyn Collect` value.
```

Comment:
It doesn't store the current value as a `dyn Collect`, that's the `next` pointer.

Reply:
I've made the comment explicitly refer to `GcBox.value`